dynamodb/src/main/scala/zio/dynamodb/DynamoDBQuery.scala

```diff
@@ -312,8 +320,8 @@ object DynamoDBQuery {
   def succeed[A](a: A): DynamoDBQuery[A] = Succeed(() => a)
 
   def forEach[A, B](values: Iterable[A])(body: A => DynamoDBQuery[B]): DynamoDBQuery[List[B]] =
-    values.foldRight[DynamoDBQuery[List[B]]](succeed(Nil)) {
-      case (a, query) => body(a).zipWith(query)(_ :: _)
+    values.foldLeft[DynamoDBQuery[List[B]]](succeed(Nil)) {
```

Comment:
I ran into a stack overflow exception and switching this to a `foldLeft` resolved it.

Reply:
you may need a `reverse` at the end of the processing chain to preserve order. 

Out of curiosity did you get the stack overflow exception when calling this function directly with a very large collection?

Reply:
This failed on 26 items which seemed rather small to overflow (granted the 26 still would've failed when the request got to Dynamo). I'll take a look at reversing tonight.

Reply:
I think I must have mixed up the cases, this definitely does not overflow with 26 items.